### DISCUSSION TITLE
<class 'OSError'> __syscall_getsockname 3

### DISCUSSION THREAD

**Discussion by tekashifuta** [pyscript/pyscript#1305] [Q&A] [answered]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help

### What happened?

Can someone help me with this.
I have been troubleshooting for the past week but no avail.

Thank You.

PS: 
I am using 
[PYZK ](https://github.com/fananimi/pyzk)
[Flask](https://flask.palletsprojects.com/en/2.2.x/quickstart/#rendering-templates)

Here's the code.
```
<!DOCTYPE html>
<html lang="en">

<head>
  <link rel="stylesheet" href="https://pyscript.net/latest/pyscript.css" />
  <script defer src="https://pyscript.net/latest/pyscript.js"></script>
  <link href='/static/pico-master/css/pico.min.css' rel="stylesheet">
  <meta charset="UTF-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>{{ title }}</title>

</head>

<body class="m-0 border-0">
  <py-config src="../static/custom.toml">
    [[fetch]]
    files = ['__init__.py', 'attendance.py', 'base.py', 'const.py', 'exception.py', 'finger.py', 'user.py']
    from = '/static/zk/'
    to_folder = 'cus_pyzk'
  </py-config>
  <py-script>
    from cus_pyzk import ZK
    
    def connectToZkteco():
      conn = None
      zk = ZK('192.168.10.123', port=4370, timeout=5, password=0, force_udp=False, ommit_ping=False)
      try:
          conn = zk.connect()
          print("Connected")

      except Exception as e:
          print('ROOOOOOOO')
          print(f'ERROR: {e.args}')
          print(f'ERROR: {type(e)}')

      finally:
          if conn:
              conn.disconnect()

    connectToZkteco()
  </py-script>
  {% include 'navbar.html' %}
  <main class="container">{% block content %} {% endblock %}</main>
</body>

</html>
```

### What browsers are you seeing the problem on? (if applicable)

Microsoft Edge

### Additional Context

OUTPUT
![error](https://user-images.githubusercontent.com/62860537/227452846-e2f5af21-8f39-4d7f-9676-0c7d0090665b.PNG)
![error2](https://user-images.githubusercontent.com/62860537/227453156-ec3b3080-144d-420c-baa0-6e9bf5974168.PNG)

**Comment by JeffersGlass** [comment, marked as answer]
It appears that that [PYZK uses the socket module](https://github.com/fananimi/pyzk/blob/89e09e31adb2907a17b13304e50d5abafbc1cad0/zk/base.py#L4) to make it's connections (to a timeclock system or similar, I'm guessing). The socket module is one of the [modules not fully supported/supportable in Pyodide](https://pyodide.org/en/stable/usage/wasm-constraints.html#included-but-not-working-modules) due to limitations within the browser-window environment - the browser simply doesn't provide access to the same kind of low-level networking primitives that a native OS does.

One potential way around this, though not a simple one, would be to re-implement the connection using a [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API), the in-browser equivalent of a socket. Perhaps by subclassing the [ZK Class](https://github.com/fananimi/pyzk/blob/89e09e31adb2907a17b13304e50d5abafbc1cad0/zk/base.py#L109) or similar? 

Since this is not a PyScript-specific issue, I'm going to convert it to a Discussion.

**Reply by tekashifuta** [reply]
Thanks @JeffersGlass
